test/functorch/test_aotdispatch.py

```diff
@@ -1723,16 +1723,32 @@ def forward(self, x):
         graph = tracer.trace(MockModule(fake_y))
         mod_fake = torch.fx.GraphModule(tracer.root, graph)
 
-        self.assertExpectedRaisesInline(
-            AssertionError, lambda: aot_module_simplified(mod_fake, (real_x,), nop),
-            """Unexpected fake buffer y"""
-        )
+        if debug:
+            inner_message = "FAKE TENSOR CREATION TRACEBACK:"
+        else:
+            inner_message = "Enable TORCH_FAKE_TENSOR_DEBUG=1 to get creation stack traces on fake tensors."
+
+        message = f"""Unexpected fake buffer y {inner_message}"""
+
+        with self.assertRaisesRegex(
+            AssertionError, message
+        ):
+            aot_module_simplified(mod_fake, (real_x,), nop)
```

Comment:
There doesn't seem to be any good reason to explicitly regex for the error message? I'd be OK with either (1) converting this two a regex but without matching the inner message or (2) having the expect test directly include Enable TORCH_FAKE_TENSOR_DEBUG=1 to get creation stack traces on fake tensors (with the expectation that the test will fail if TORCH_FAKE_TENSOR_DEBUG=1 is true. You could work around that by setting up a little post process function to strip out the backtrace though.). Simultaneously converting it to a regex AND regexing for the inner message seems like the worst of both worlds.

Reply:
Why is it the worst of both worlds? the point is to check if the assertion is produced as expected, which this does.

Reply:
It's the worst of both worlds because (1) you are no longer using assertExpectedInline, so if the message changes, you are going to have to go manually update the test, as opposed to getting automated updates with expect test, but (2) you have made an extremely specific error message test (including the help text), which makes it extremely likely that if the message wobbles at all, you will have to update the test. You have essentially guaranteed that a manual update is necessary if any aspect of this error reporting changes.